kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/ConsoleOptions.java

```diff
@@ -271,4 +273,40 @@ private void addAuthenticationMethodProperties() {
         }
     }
 
+    @Override
+    public int hashCode() {
+        return Objects.hash(this.allowedPorts, this.appRoot, this.authenticationMethodProperties, this.bannerContent,
+                this.bannerEnabled, this.config, this.configurationProperties, this.passwordMinLength,
+                this.passwordRequireBothCases, this.passwordRequireDigits, this.passwordRequireSpecialCharacters,
+                this.sessionMaxInactivityInterval, this.sslManagerServiceTarget);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || getClass() != obj.getClass()) {
+            return false;
+        }
+        ConsoleOptions other = (ConsoleOptions) obj;
+
+        return Objects.equals(this.appRoot.get(), other.appRoot.get())
```

Comment:
Should we compare the result of getOptional() instead of get() where possible? If for some reason the property value is not set get() will throw

Reply:
from what I see appRoot has the default value (as do the other fields, except allowedPorts) so it is impossible, in this case, for the get method to throw the exception.